spring-integration-core/src/main/java/org/springframework/integration/support/management/observation/IntegrationObservation.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.support.management.observation;
+
+import io.micrometer.common.docs.KeyName;
+import io.micrometer.observation.docs.DocumentedObservation;
+
+/**
+ * The {@link DocumentedObservation} implementation for Spring Integration infrastructure.
+ *
+ * @author Artem Bilan
+ *
+ * @since 6.0
+ */
+public enum IntegrationObservation implements DocumentedObservation {
+
+	/**
+	 * Observation for message handlers.
+	 */
+	HANDLER {
+		@Override
+		public String getName() {
+			return "spring.integration.handler";
+		}
+
+		@Override
+		public String getPrefix() {
+			return "spring.integration.";
+		}
+
+		@Override
+		public Class<DefaultMessageReceiverObservationConvention> getDefaultConvention() {
+			return DefaultMessageReceiverObservationConvention.class;
+		}
+
+		@Override
+		public KeyName[] getLowCardinalityKeyNames() {
+			return HandlerTags.values();
+		}
+
+	};
+
+	/**
+	 * Key names for message handler observations.
+	 */
+	enum HandlerTags implements KeyName {
+
+		/**
+		 * Name of the message handler component.
+		 */
+		COMPONENT_NAME {
+			@Override
+			public String asString() {
+				return "spring.integration.name";
```

Comment:
Shouldn't this be `spring.integration.handler.name` ? And below, `spring.integration.handler.type` ?

Reply:
No, I don't think so.
It would sound a bit awkward: `spring.integration.handler.type=handler`.

The idea is to have a common tag on integration meters, so we can easy search them.
There are going to be also: `spring.integration.type=source` and `spring.integration.type=channel`.
The `spring.integration.name` really belongs to the bean name of that integration component.

Not sure why would you request me to do that fine-graining...

Reply:
Ah, ok, I misunderstood what the type is. I thought that you would have `spring.integration.handler.type=some handler type`. Disregard my comment.